pkg/api/tenant.go

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/grafana/dskit/middleware"
+	"github.com/grafana/dskit/tenant"
+)
+
+const (
+	tooManyTenantsTemplate = "too many tenant IDs present. max: %d actual: %d"
+)
+
+// NewTenantValidationMiddleware creates a new middleware that validates the number of tenants
+// being accessed in a particular request is allowed given the current tenant federation configuration.
+// Note that this middleware requires that tenant ID has been set on the request context by something
+// like middleware.AuthenticateUser.
+func NewTenantValidationMiddleware(federation bool, maxTenants int) middleware.Interface {
```

Comment:
do we need this exported?

Reply:
Nope, will fix.

Reply:
Fixed in fbd4049b0b3d447e4afb960e8826ee4cb818b0cd